src/widget-core/WidgetBase.ts

```diff
@@ -52,15 +52,38 @@ function isDomMeta(meta: any): meta is Base {
 	return Boolean(meta.afterRender);
 }
 
-const IGNORE_LIST: (string | symbol)[] = ['constructor', 'render'];
+const IGNORE_LIST: (string | symbol)[] = ['render', ...Object.getOwnPropertyNames(Object.getPrototypeOf({}))];
+
+const autoBindCache = new Map<any, string[]>();
 
 function autoBind(instance: any) {
-	let keys: string[] = Object.getOwnPropertyNames(instance.constructor.prototype);
+	let prototype = instance.constructor.prototype;
+
+	let keys: string[] = [];
+
+	if (autoBindCache.has(prototype)) {
+		keys = autoBindCache.get(prototype) as string[];
+	} else {
+		let p = prototype;
+		while (p) {
+			const ownKeys = Object.getOwnPropertyNames(p);
+
+			if (ownKeys.indexOf('_type') !== -1) {
+				break;
+			}
+
+			keys = [...keys, ...ownKeys];
+			p = Object.getPrototypeOf(p);
+		}
+
+		keys = keys.filter((k) => typeof instance[k] === 'function' && IGNORE_LIST.indexOf(k) === -1);
```

Comment:
The filtering of keys through the ignore list is now done up here rather than down yonder.